lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoders.java

```diff
@@ -534,14 +544,33 @@ public ValueDecoder<long[]> getInt64TimestampMicrosToShortTimestampDecoder(Parqu
         int precision = ((TimestampType) field.getType()).getPrecision();
         ValueDecoder<long[]> valueDecoder = getLongDecoder(encoding);
         if (precision == 6) {
-            return valueDecoder;
+            if (timeZone == DateTimeZone.UTC) {
+                return valueDecoder;
+            }
+            new InlineTransformDecoder<>(
+                    valueDecoder,
+                    (values, offset, length) -> {
+                        // decoded values are epochMicros, round to lower precision
```

Comment:
Comment doesn't make sense, the precision is already 6 - there's no lower precision.